src/platform/packages/shared/kbn-scout/src/cli/config_discovery.test.ts

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import fs from 'fs';
+import { FlagsReader } from '@kbn/dev-cli-runner';
+import { ToolingLog } from '@kbn/tooling-log';
+import { getScoutPlaywrightConfigs } from '../config';
+import { runDiscoverPlaywrightConfigs } from './config_discovery';
+import { measurePerformance } from '../common';
+import { validateWithScoutCiConfig } from '../config/discovery';
+
+jest.mock('fs');
+
+jest.mock('@kbn/scout-info', () => ({
+  SCOUT_PLAYWRIGHT_CONFIGS_PATH: '/path/to/scout_playwright_configs.json',
+}));
+
+jest.mock('../common', () => ({
+  measurePerformance: jest.fn(),
+}));
+
+jest.mock('../config', () => ({
+  getScoutPlaywrightConfigs: jest.fn(),
+}));
+
+jest.mock('../config/discovery', () => ({
+  validateWithScoutCiConfig: jest.fn(),
+}));
+
+describe('runDiscoverPlaywrightConfigs', () => {
+  let flagsReader: jest.Mocked<FlagsReader>;
+  let log: jest.Mocked<ToolingLog>;
+
+  // 'enabled' plugins
+  const mockPluginsWithRunnableConfigs: Map<string, any> = new Map([
+    [
+      'pluginA',
+      {
+        group: 'groupA',
+        pluginPath: 'plugin/path',
+        configs: ['config1.ts', 'config2.ts'],
+        usesParallelWorkers: true,
+      },
+    ],
+  ]);
+
+  // these are all the plugins found, but only some of them will be enabled (validateWithScoutCiConfig will find out which ones)
+  const mockPluginsWithConfigs = new Map([
+    ['pluginA', { group: 'groupA', configs: ['config1.ts', 'config2.ts'] }], // enabled
+    ['pluginB', { group: 'groupB', configs: ['config3.ts'] }], // disabled
+  ]);
+
+  beforeAll(() => {
+    flagsReader = {
+      arrayOfStrings: jest.fn(),
+      boolean: jest.fn(),
+    } as any;
+
+    log = {
+      info: jest.fn(),
+      error: jest.fn(),
+      warn: jest.fn(),
+    } as any;
+  });
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+
+    (fs.existsSync as jest.Mock).mockReturnValue(false);
+    (fs.mkdirSync as jest.Mock).mockImplementation(jest.fn());
+    (fs.writeFileSync as jest.Mock).mockImplementation(jest.fn());
+
+    (measurePerformance as jest.Mock).mockImplementation((_log, _msg, fn) => fn());
+
+    (getScoutPlaywrightConfigs as jest.Mock).mockReturnValue(mockPluginsWithConfigs);
+    (validateWithScoutCiConfig as jest.Mock).mockReturnValue(mockPluginsWithRunnableConfigs);
+  });
+
+  it('validates configs when "validate" is true', () => {
+    // force --validate
+    flagsReader.boolean.mockImplementation((flag) => (flag === 'validate' ? true : false));
```

Comment:
Since we might add more flags in the future, maybe we better define all input flags explicitly?

Reply:
Good idea! Implemented in [0de79b7](https://github.com/elastic/kibana/pull/216272/commits/0de79b79aed8c01138dd959fdb908494e773227f)